src/Emu/Metadata/SupportFiles/FrontierLabs/LogFile.cs

```diff
@@ -128,16 +134,40 @@ public static LogFile GetLogFile(string log)
             return null;
         }
 
+        public static LocalDateTime ParseDate(string dateTime)
+        {
+            LocalDateTime timeStamp;
+
+            foreach (LocalDateTimePattern datePattern in DatePatterns)
+            {
+                try
+                {
+                    timeStamp = datePattern.Parse(dateTime).Value;
+                    return timeStamp;
+                }
+                catch (UnparsableValueException)
+                {
+                    continue;
+                }
+            }
+
+            throw new UnparsableValueException($"Can't parse date {dateTime}");
```

Comment:
There's no need to throw and catch exceptions here. Check the Parse result's `Success` property.